### ISSUE TITLE
Option to display Last Run and Next Run based on data_interval_end

### ISSUE THREAD

**Issue by fatmumuhomer** [apache/airflow#22238]
### Description

Currently the DAGs view shows the Last Run and Next Run values using the data_interval_start. The full data interval information is available if the user hovers over the field. Airflow should allow these two fields to display the data_interval_end instead. This could be configurable at a system level via an environment variable or airflow.cfg or, if possible, per user.

### Use case/motivation

There are two use case:
1. To support users who are interested in monitoring data loads but who are not familiar with Airflow and may be confused by the concept of the data interval and, specifically, why the UI is showing the data_interval_start.
2. The majority of the DAGs my team manages care more about the data_interval_end than the start and so we would, at a glance, prefer to be able to see the DAG information in relation to that.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
We were talking about switching the last run and next run to be based off of interval end instead: https://github.com/apache/airflow/issues/21493

**Comment by fatmumuhomer** [comment]
Thanks @bbovenzi. From the discussion it sounds like the issue you linked would take care of my concern as it would cause those two columns to always show the value based on data_interval_end. Is this something you are actively working on?

**Comment by bbovenzi** [comment]
No, feel free to pick it up! I'm working on some adjacent things but there shouldn't be any conflicts.

**Comment by fatmumuhomer** [comment]
Great! Thanks! I will take a look at the code first before I grab the issue.

**Comment by caupetit-itf** [comment]
Hello ! I would be very interested by this feature to open the UI to some stakeholders, is this still an ongoing thing ?

**Comment by fatmumuhomer** [comment]
It's something I'm figuring out how to implement. I still think it's a good change to make as it regularly causes confusion for users at my company.

**Comment by uranusjr** [comment]
Now that we move the ordering logic to the timetable (#25633 and #25880), we should display this information based on that.

**Comment by fatmumuhomer** [comment]
@uranusjr Will that also work for DAGs that are scheduled using other methods such as cron expressions?

**Comment by bbovenzi** [comment]
In Airflow 3 and the loss of a unique logical_date, we now default sort by `run_after`
